content/plugin/framework/resources/import.mdx

```diff
@@ -0,0 +1,109 @@
+        },
+    }, nil
+}
+```
+
+Along with a `tfsdk.Resource` implementation with the following `Read` method:
+
+```go
+func (r exampleResource) Read(ctx context.Context, req tfsdk.ReadResourceRequest, resp *tfsdk.ReadResourceResponse) {
+    var attrOne, attrTwo string
+
+    resp.Diagnostics.Append(req.State.GetAttribute(ctx, tftypes.NewAttributePath().WithAttributeName("attr_one"), &attrOne)...)
+    resp.Diagnostics.Append(req.State.GetAttribute(ctx, tftypes.NewAttributePath().WithAttributeName("attr_two"), &attrTwo)...)
+
+    if resp.Diagnostics.HasError() {
+        return
+    }
+
+    // API call using attrOne and attrTwo
+}
+```
+
+The `terraform import` command will need to accept both attribute values as a single import identifier string. A typical convention is to use a separator character, such as a comma (`,`), between the values. The `ImportState` method will then need to parse the import identifier string into the two separate values and save them appropriately into the Terraform state.
+
+Using a comma separated value, the `ImportState` method can have the following:
+
+```go
+func (r exampleResource) ImportState(ctx context.Context, req tfsdk.ImportResourceStateRequest, resp *tfsdk.ImportResourceStateResponse) {
+    idParts := strings.Split(req.ID, ",")
+
+    if len(idParts) != 2 || idParts[0] == "" || idParts[1] == "" {
+        resp.Diagnostics.AddError(
+            "Unexpected Import Identifier",
+            fmt.Sprintf("Expected import identifier with format: attr_one,attr_two. Got: %q", req.ID),
+        )
+        return
+    }
+
+    resp.Diagnostics.Append(resp.State.SetAttribute(ctx, tftypes.NewAttributePath().WithAttributeName("attr_one"), idParts[0])...)
+    resp.Diagnostics.Append(resp.State.SetAttribute(ctx, tftypes.NewAttributePath().WithAttributeName("attr_two"), idParts[1])...)
+}
+```
+
+## Not Implemented
+
+The [`tfsdk.ResourceImportStateNotImplemented` function](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/tfsdk#ResourceImportStateNotImplemented) is available to return an error to practitioners if the resource does not implement import support. For example:
+
+```go
+func (r exampleResource) ImportState(ctx context.Context, req tfsdk.ImportResourceStateRequest, resp *tfsdk.ImportResourceStateResponse) {
+    tfsdk.ResourceImportStateNotImplemented(ctx, "", resp)
+}
+```
+
+When a practitioner attempts to run the `terraform import` command against a resource with this implementation, Terraform CLI will return this default messaging:
+
+```console
+$ terraform import example_resource.example some-identifier
+example_resource.example: Importing from ID "some-identifier"...
+╷
+│ Error: Resource Import Not Implemented
+│ 
+│ This resource does not support import. Please contact the provider developer for additional information.
+╵
+```
+
+Modifying the details parameter of the function allows for customized recommendations, such as other actions the practitioner can take.
```

Comment:
I can definitely see the potential for confusion here! The function signature names the parameter `details` to align with the concept that it is changing the "details" section of the error diagnostic it is creating. I wonder if it might be slightly clearer to further mention replacing the `""` in the above example.

Reply:
FWIW - I like the idea you just had :-)